sycl/test/regression/macro_conflict.cpp

```diff
@@ -0,0 +1,18 @@
+// RUN: %clangxx -fsyntax-only -Xclang -verify %s -o %t.out
+// expected-no-diagnostics
+//
+//===----------------------------------------------------------------------===//
+// This test checks if the user-defined macros SUCCESS, FAIL, BLOCKED are
+// defined in global namespace by sycl.hpp
+//===----------------------------------------------------------------------===//
+
+#define SUCCESS 0
```

Comment:
Are you sure we are suppose to fix this type of issues in our headers?
What if user defines something like:
```c++
#define i j
```
?
I would prefer user to this this problem by defining macro after SYCL header includes.

Reply:
There is no way for us to force the user to define macros afteer sycl.hpp.
As I mentioned before, we cannot avoid user macro conflict 100%.
Preprocessor will change the source code with user defined macros.
We only need to use unlikely variables to minimize the chance.